domain/secret/state/state.go

```diff
@@ -2339,9 +2339,79 @@ AND    subject_type_id != $M.remote_application_type`
 	return accessors.toSecretGrants(accessScopes)
 }
 
+// InitialWatchStatementForConsumedSecrets returns the initial watch statement and the table name for watching consumed secrets.
+func (st State) InitialWatchStatementForConsumedSecrets(unitName string) (string, eventsource.NamespaceQuery) {
+	queryFunc := func(ctx context.Context, runner database.TxnRunner) ([]string, error) {
+		return st.getConsumedSecretURIs(ctx, runner, unitName)
+	}
+	return "secret_unit_consumer", queryFunc
+}
+
+// GetConsumedSecretURIs returns the URIs of the secrets consumed by the specified unit.
+func (st State) GetConsumedSecretURIs(ctx context.Context, unitName string, consumerIDs ...string) ([]string, error) {
+	db, err := st.DB()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return st.getConsumedSecretURIs(ctx, db, unitName, consumerIDs...)
+}
+
+type secretUnitConsumerIDs []string
+
+func (st State) getConsumedSecretURIs(
+	ctx context.Context, runner coredatabase.TxnRunner, unitName string,
+	consumerIDs ...string,
+) ([]string, error) {
+	q := `
+SELECT suc.secret_id AS &secretUnitConsumer.secret_id
+FROM secret_unit_consumer suc
+LEFT JOIN unit u ON u.uuid = suc.unit_uuid
```

Comment:
This should just be an INNER join right? There's a FK from unit_uuid to unit table.